pyccel/codegen/printing/cwrappercode.py

```diff
@@ -330,6 +333,193 @@ def optional_element_management(self, used_names, a, collect_var):
+                mini_wrapper_func_vars[r.name] = r
+                if cast_func is not None:
+                    mini_wrapper_func_body.append(cast_func)
+                    mini_wrapper_func_vars[collect_var.name] = collect_var
+                res_args.append(VariableAddress(collect_var) if collect_var.is_pointer else collect_var)
+
+            # Building PybuildValue and freeing the allocated variable after.
+            mini_wrapper_func_body.append(AliasAssign(wrapper_results[0],PyBuildValueNode(res_args)))
+            mini_wrapper_func_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+            mini_wrapper_func_body.append(Return(wrapper_results))
+            self._to_free_PyObject_list.clear()
+
+            # Building Mini wrapper function
+            mini_wrapper_func_name = self.get_new_name(used_names.union(self._global_names), func.name.name + '_mini_wrapper')
+            self._global_names.add(mini_wrapper_func_name)
+
+            mini_wrapper_func_def = FunctionDef(name = mini_wrapper_func_name,
+                arguments = parse_args,
+                results = wrapper_results,
+                body = mini_wrapper_func_body,
+                local_vars = mini_wrapper_func_vars.values())
+            funcs_def.append(mini_wrapper_func_def)
+
+            # append check condition to the functioncall
+            body_tmp.append((PyccelEq(check_var, LiteralInteger(flags)), [AliasAssign(wrapper_results[0],
+                    FunctionCall(mini_wrapper_func_def, parse_args))]))
+
+        # Errors / Types management
+        # Creating check_type function body
+        check_func_body = []
+        flags = (len(types_dict) - 1) * 4
+        for a in types_dict:
+            body = []
+            types = []
+            for s in types_dict[a] :
+                value = s[2] << flags
+                body.append((s[1], [AugAssign(check_var, '+' ,value)]))
+                types.append(s[0].dtype)
+            flags -= 4
+            error = ' or '.join([str_dtype(v) for v in types])
+            body.append((LiteralTrue(), [PyErr_SetString('PyExc_TypeError', '"{} must be {}"'.format(types_dict[a][0].name, error)), Return([LiteralInteger(0)])]))
+            check_func_body += [If(*body)]
+        check_func_body = [Assign(check_var, LiteralInteger(0))] + check_func_body
+        check_func_body.append(Return([check_var]))
+        # Creating check function definition
+        check_func_name = self.get_new_name(used_names.union(self._global_names), 'type_check')
+        self._global_names.add(check_func_name)
+        check_func_def = FunctionDef(name = check_func_name,
+            arguments = parse_args,
+            results = [check_var],
+            body = check_func_body,
+            local_vars = [])
+        funcs_def.append(check_func_def)
+
+        # Create the wrapper body with collected informations
+        body_tmp = [((PyccelNot(check_var), [Return([Nil()])]))] + body_tmp
```

Comment:
`if ( ! check_var ) return NILL` ?
Shouldn't this have an associated error message?

Reply:
Here : https://github.com/pyccel/pyccel/pull/548/files#diff-69e3bda8e13a07d02bdfe586b58c9b581dd6497ea648cbe202e535f31c06c7aaR458
?
:+1:

Reply:
yes